aws/retry/middleware.go

```diff
@@ -87,10 +90,19 @@ func (r Attempt) HandleFinalize(ctx context.Context, in smithymiddle.FinalizeInp
 			AttemptClockSkew: attemptClockSkew,
 		})
 
+		var attemptMetadata requestAttempt
+		defer func() {
+			allAttemptMetadata.Attempts = append(allAttemptMetadata.Attempts, attemptMetadata)
+		}()
+
 		if attemptNum > 1 {
+			attemptMetadata.Retried = true
+
```

Comment:
Won't this refer to the previous attempt begin retried not this current attempt being retried?  I'd expect the `Retried` member to be set on the previous attempt not the current attempt.

consider moving setting `Retried` until after the check if will retry or not, e.g. line 140.